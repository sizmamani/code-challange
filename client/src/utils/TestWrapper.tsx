import { render } from '@testing-library/react';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';
import { ThemeProvider } from 'styled-components';
import bitTheme from '../theme/bitTheme';
import { GRAPHQL_API_URL } from '../config';

const client = new ApolloClient({
    uri: GRAPHQL_API_URL,
});

function withProvider(ui: any) {
    const Wrapper = (props: any) => (
        <ApolloProvider client={client as any}>
            <ThemeProvider theme={bitTheme}>{props.children}</ThemeProvider>
        </ApolloProvider>
    );
    return {
        ...render(ui, { wrapper: Wrapper }),
    };
}

export * from '@testing-library/react';
export { withProvider };


