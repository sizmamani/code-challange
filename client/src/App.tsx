import { Route, Router, Switch } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { TopHeader } from './components/header';
import { BlockDetailPage, BlockListingPage } from './pages';
import bitTheme from './theme/bitTheme';
import { history } from './utils/history';

const App = () => (
  <ThemeProvider theme={bitTheme}>
    <TopHeader />
    <Router history={history}>
      <Switch>
        <Route path='/:hash' component={BlockDetailPage} />
        <Route path='/' component={BlockListingPage} />
      </Switch>
    </Router>
  </ThemeProvider>
);

export default App;
