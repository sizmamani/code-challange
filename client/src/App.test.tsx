import App from './App';
import { withProvider } from './utils/TestWrapper';

it('Should render withProvider and match the snapshot', () => {
  const {
    container: { firstChild },
  } = withProvider(<App />);
  expect(firstChild).toMatchSnapshot();
});
