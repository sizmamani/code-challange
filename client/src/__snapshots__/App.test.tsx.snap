// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render WithTheme 1`] = `
<div
  class="sc-bdnxRM cdLDeN"
>
  <img
    alt="logo"
    src="logo.png"
    width="87"
  />
</div>
`;

exports[`Should render WithTheme and Appo 1`] = `
<div
  class="sc-bdnxRM cdLDeN"
>
  <img
    alt="logo"
    src="logo.png"
    width="87"
  />
</div>
`;

exports[`Should render withProvider and match the snapshot 1`] = `
<div
  class="sc-bdnxRM cdLDeN"
>
  <img
    alt="logo"
    src="logo.png"
    width="87"
  />
</div>
`;
